src/Language/Mulang/Inspector/Procedural.hs

```diff
@@ -45,3 +46,6 @@ usesForLoop :: Inspection
 usesForLoop = containsExpression f
   where f (ForLoop _ _ _ _) = True
         f _                 = False
+
+usesLoop :: Inspection
+usesLoop e = usesRepeat e || usesWhile e || usesForLoop e || usesForEach e
```

Comment:
Just a side note for the future: this implementation does not perform well, since it traverses the tree several times, instead of just once.